### PR TITLE
Linting configuration

### PR DESCRIPTION
**Pull request by sam-drumm**
https://hackney.atlassian.net/browse/TS-1586?atlOrigin=eyJpIjoiOGYzNjA3OTNjYjIwNGY2OWI5YWM0NWE2OTk4YzgwYzEiLCJwIjoiaiJ9

**What**
- Adds linter configuration from Hackney and necessary libraries, 
- Adds a precommit step to check tests, tsc, lint and prettify staged files
- Adds coverage reports for Jest so that CircleCI can show results and Sonar can run analysis
- Adds .nvmrc 

**Why**
As linting rules are not applied to this directory, we will have a lot of errors and warnings to work through if we apply it to the whole code base. 

This PR allows us to have better feedback in the IDE, but only force rules on staged files. 

**How**
- Husky will run a precommit step to lint-fix/ prettify-fix staged files, run tests, run tsc. Commit will fail if not passed.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  },
+  "ignorePatterns": ["mocks/**", "coverage/**"],
+  "rules": {
+    "linebreak-style": ["error", "unix"],
+    "require-atomic-updates": "off",
+    "react/react-in-jsx-scope": "off",
+    // "react/function-component-definition": ["warn"],
+    // "react/jsx-no-useless-fragment": ["warn"],
+    // "jsx-a11y/anchor-is-valid": ["warn"],
+    // "eqeqeq": ["warn", "smart"],
```
Comment (Duslerke) [inline .eslintrc:34]: perhaps, remove if commented?

File: package-lock.json
Comment (Duslerke) [inline package-lock.json:1]: What happened here?

**Review by Duslerke — APPROVED** [review]

**Comment by Duslerke** [comment]
Approved, but I recommend resolving the package lock diff now

File: package-lock.json
Comment (sam-drumm) [reply package-lock.json:1]: Yeah good catch didn't notice that one thanks

Comment (sam-drumm) [reply package-lock.json:1]: Interesting that I can push commits to this branch after approval and it doesn't go stale....

File: .circleci/config.yml [hunk 2]
```diff
@@ -131,7 +141,7 @@ workflows:
       - sonar-scan:
           context: SonarCloud
           requires:
-            - install-dependencies
+            - tests
```
Comment (sam-drumm) [inline .circleci/config.yml:144]: Requires test rather than in parallel because it relies on the coverage reports.

**Comment by sam-drumm** [comment]
I understand this PR would make this.. 

https://github.com/LBHackney-IT/lbh-housing-register/blob/TS-1586-Housing-reg-view-only-Review-linting/.pre-commit-config.yaml

...redundant - but want to review that.

**Comment by LBHTKarki** [comment]
Now that we are using husky I guess we could remove the pre-commit config file `.pre-commit-config.yaml`

**Review by LBHTKarki — APPROVED** [review]

**Pull request merged**